instrumentation/httpclient5/src/main/java/brave/httpclient5/TracingHttpClients.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright ${license.git.copyrightYears} The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package brave.httpclient5;
+
+import brave.Tracing;
+import brave.http.HttpTracing;
+import brave.propagation.CurrentTraceContext;
+import org.apache.hc.client5.http.impl.ChainElement;
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.client5.http.impl.async.H2AsyncClientBuilder;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClientBuilder;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+
+import static org.apache.hc.client5.http.impl.ChainElement.PROTOCOL;
+
+public class TracingHttpClients {
+  public static CloseableHttpClient create(Tracing tracing, HttpClientBuilder httpClientBuilder) {
```

Comment:
separate duty of creating HttpClient5Tracing component from making a CloseableHttpClient. this allows one-time configuration even if using multiple components.

ex.
```java
HttpClient5Tracing httpClient5Tracing = HttpClient5Tracing.newBuilder(httpTracing);

httpAsyncClient = httpClient5Tracing.build(httpAsyncClientBuilder);
```

Reply:
Curious usually `HttpTracing` has the same server name, so the same instance should be used for the same endpoint? But separate to `newBuilder` & `build` still better than a single call.